File: 0162-find-peak-element/0162-find-peak-element.cpp
class Solution {
public:
    int findPeakElement(vector<int>& nums) {
        int left, right, mid, length;
        long left_num, right_num, mid_num;
        length = nums.size();
        left = 0;
        right = length;

        while(true){
            mid = (left+right)/2;
            if (mid > 0){
                left_num = nums[mid-1];
            }else{
                left_num = numeric_limits<long>::min();
            }

            if (mid < length-1){
                right_num = nums[mid+1];
            }else{
                right_num = numeric_limits<long>::min();
            }

            mid_num = nums[mid];

            if (left_num < mid_num && right_num < mid_num){
                return mid;
            }
            else if (left_num > mid_num){
                right = mid;
            }
            else{
                left = mid+1;
            }
        }
        
    }
};

File: 0162-find-peak-element/0162-find-peak-element.py
class Solution:
    def findPeakElement(self, nums: List[int]) -> int:
        left = 0
        length = len(nums)
        right = length
        
        while True:
            mid = (left+right)//2
            left_num = nums[mid-1] if mid > 0 else float('-inf')
            right_num = nums[mid+1] if mid < length-1 else float('-inf')
            if left_num < nums[mid] > right_num:
                return mid
            elif left_num > nums[mid]:
                right = mid
            elif right_num > nums[mid]:
                left = mid+1